src/reason-parser/reason_parser.mly

```diff
@@ -3111,15 +3111,21 @@ expr_optional_constraint:
 record_expr:
   | DOTDOTDOT expr_optional_constraint lnonempty_list(preceded(COMMA,lbl_expr))
     { (Some $2, $3) }
-  | as_loc(label_longident) COLON expr
-    { (None, [($1, $3)]) }
+  | attribute as_loc(label_longident) COLON expr
+    { let rec_expr = {$4 with pexp_attributes = [$1] @ $4.pexp_attributes} in
+      (None, [($2, rec_expr)])
+    }
   | lseparated_two_or_more(COMMA, lbl_expr) COMMA?
     { (None, $1) }
 ;
 
 lbl_expr:
-  | as_loc(label_longident) COLON expr { ($1, $3) }
-  | as_loc(label_longident)            { ($1, exp_of_label $1) }
+  | attribute as_loc(label_longident) COLON expr
+    { ($2, {$4 with pexp_attributes = [$1] @ $4.pexp_attributes} ) }
+  | attribute as_loc(label_longident)
+    { let exp = exp_of_label $2 in
+      ($2, {exp with pexp_attributes = [$1] @ exp.pexp_attributes})
+    }
```

Comment:
This compiles but broke attributes like this
```reason
[@bs.config {foo, jsx: 2}];
```
with this error that also appears many times on other tests:
```
Error: 2823: <UNKNOWN SYNTAX ERROR>
```
and also gave
```
File "./formatTest/unit_tests/input/./basicStructures.re", line 210, characters 25-28:
Error: 304: fun is a reserved keyword, it cannot be used as an identifier. Try `fun_' instead
```
It may be worthwhile trying a different approach unless someone has other ideas on how to make this work. Trying to use `item_attributes` instead introduces shift reduce conflicts that I don't yet really know how to deal with. Since we have the 4.04 AST available potentially we could use the [Docstrings](https://caml.inria.fr/pub/docs/manual-ocaml/libref/Docstrings.html) module to handle docstrings.